src/main/java/com/squareup/catalog/demo/example/GloballyEnableAllItemsExample.java

```diff
@@ -0,0 +1,137 @@
+ * limitations under the License.
+ */
+package com.squareup.catalog.demo.example;
+
+import com.squareup.catalog.demo.Logger;
+import com.squareup.connect.ApiException;
+import com.squareup.connect.api.CatalogApi;
+import com.squareup.connect.api.LocationsApi;
+import com.squareup.connect.models.BatchUpsertCatalogObjectsRequest;
+import com.squareup.connect.models.BatchUpsertCatalogObjectsResponse;
+import com.squareup.connect.models.CatalogItem;
+import com.squareup.connect.models.CatalogItem.ProductTypeEnum;
+import com.squareup.connect.models.CatalogObject;
+import com.squareup.connect.models.CatalogObjectBatch;
+import com.squareup.connect.models.ListCatalogResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static com.squareup.connect.models.CatalogItem.ProductTypeEnum.REGULAR;
+import static com.squareup.connect.models.CatalogObjectType.ITEM;
+import static java.util.Collections.emptyList;
+
+/**
+ * This example makes all items available at all current and future locations.
+ **/
+public class GloballyEnableAllItemsExample extends Example {
+
+  public GloballyEnableAllItemsExample(Logger logger) {
+    super("globally_enable_items", "Make all items available at all locations.", logger);
+  }
+
+  @Override public void execute(CatalogApi catalogApi, LocationsApi locationsApi)
+      throws ApiException {
+    String cursor = null;
+    int totalItemsUpdated = 0;
+    int totalItemsVisited = 0;
+    final long startTimeMillis = System.currentTimeMillis();
+    do {
+      // Retrieve a page of items.
+      ListCatalogResponse listResponse = catalogApi.listCatalog(cursor, ITEM.toString());
+      if (checkAndLogErrors(listResponse.getErrors())) {
+        return;
+      }
+
+      List<CatalogObject> items = listResponse.getObjects();
+      if (items.size() == 0) {
+        if (cursor == null) {
+          logger.info("No items found in catalog.");
+          return;
+        }
+      } else {
+        // Iterate over the items, enabling them at all locations.
+        List<CatalogObject> itemsToUpdate = new ArrayList<>();
+        for (CatalogObject itemObject : items) {
+          CatalogItem item = itemObject.getItemData();
+          ProductTypeEnum productType = item.getProductType();
+          boolean isRegularItem = productType == null || productType == REGULAR;
+          boolean isGloballyEnabled = itemObject.getPresentAtAllLocations()
+              && itemObject.getPresentAtLocationIds().isEmpty()
+              && itemObject.getAbsentAtLocationIds().isEmpty();
+
+          // Ignore items that are not REGULAR or are already globally enabled.
+          if (isRegularItem && !isGloballyEnabled) {
+            // Globally enable the item.
+            itemObject.setPresentAtAllLocations(true);
+            itemObject.setPresentAtLocationIds(emptyList());
+            itemObject.setAbsentAtLocationIds(emptyList());
+
```

Comment:
In Connect v2, the item image is represented as a simple URL string. Connect V2 automatically handles updating the enabled locations of the image object on the backend.

Item variations are different because we have first class support for disabling item variations at specified locations.